src/main/java/gregtech/common/metatileentities/converter/ConverterTrait.java

```diff
@@ -0,0 +1,268 @@
+
+    private long usedAmps;
+
+    private BlockPos frontPos;
+
+    public ConverterTrait(MetaTileEntityConverter mte, int amps, boolean feToEu) {
+        super(mte);
+        this.amps = amps;
+        this.feToEu = feToEu;
+        this.voltage = GTValues.V[mte.getTier()];
+        this.baseCapacity = this.voltage * 16 * amps;
+    }
+
+    protected IEnergyContainer getEnergyEUContainer() {
+        return energyEU;
+    }
+
+    protected IEnergyStorage getEnergyFEContainer() {
+        return energyFE;
+    }
+
+    public boolean isFeToEu() {
+        return feToEu;
+    }
+
+    protected void setFeToEu(boolean feToEu) {
+        this.feToEu = feToEu;
+    }
+
+    public int getBaseAmps() {
+        return amps;
+    }
+
+    public long getVoltage() {
+        return voltage;
+    }
+
+    @Override
+    public String getName() {
+        return "EnergyConvertTrait";
+    }
+
+    @Override
+    public int getNetworkID() {
+        return 1;
+    }
+
+    @Override
+    public <T> T getCapability(Capability<T> capability) {
+        return null;
+    }
+
+    private long extractInternal(long amount) {
+        if (amount <= 0) return 0;
+        long change = Math.min(storedEU, amount);
+        storedEU -= change;
+        return change;
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT() {
+        NBTTagCompound nbt = new NBTTagCompound();
+        nbt.setLong("StoredEU", storedEU);
+        nbt.setBoolean("feToEu", feToEu);
+        return nbt;
+    }
+
+    @Override
+    public void deserializeNBT(NBTTagCompound nbt) {
+        this.storedEU = nbt.getLong("StoredEU");
+        this.feToEu = nbt.getBoolean("feToEu");
+    }
+
+    @Override
+    public void update() {
+        super.update();
+        this.usedAmps = 0;
+        if (!metaTileEntity.getWorld().isRemote) {
+            pushEnergy();
+        }
+    }
+
+    protected void pushEnergy() {
+        long energyInserted;
+        if (feToEu) { // push out EU
+            // Get the EU capability in front of us
+            IEnergyContainer container = getCapabilityAtFront(GregtechCapabilities.CAPABILITY_ENERGY_CONTAINER);
+            if (container == null) return;
+
+            // make sure we can output at least 1 amp
+            long ampsToInsert = Math.min(amps, storedEU / voltage);
+            if (ampsToInsert == 0) return;
+
+            // send out energy
+            energyInserted = container.acceptEnergyFromNetwork(metaTileEntity.getFrontFacing().getOpposite(), voltage, ampsToInsert) * voltage;
+        } else { // push out FE
+            // Get the FE capability in front of us
+            IEnergyStorage storage = getCapabilityAtFront(CapabilityEnergy.ENERGY);
+            if (storage == null) return;
+
+            // send out energy
+            int feToSend = FeCompat.toFe(storedEU - (storedEU % FeCompat.ratio(false))); // don't lose EU precision
+            energyInserted = FeCompat.toEu(storage.receiveEnergy(feToSend, false));
```

Comment:
Not sure what's up with the first line.
Second line will void EU, int division.